array.go

```diff
@@ -25,6 +25,19 @@ func NewArray(reporter Reporter, value []interface{}) *Array {
 	return newArray(newChainWithDefaults("Array()", reporter), value)
 }
 
+// NewArrayC returns a new Array instance with config
+//
+// Both config and value should not be nil. If value is nil, failure is
+// reported.
+//
+// Example:
+//
+// array := NewArrayC(conf,[]interface{}{"foo",123})
```

Comment:
Please ensure that all code example are correctly indented.

There should be a tab (not a space and not a few spaces) right after `//`.

There should be also an empty line right after `Example:`.

E.g., this one is GOOD:

![image](https://user-images.githubusercontent.com/8436629/209313369-19872b6d-758b-4a56-bbda-d59a795f8a91.png)

Also, there should NOT be an empty line between comment and `func`, like here:

![image](https://user-images.githubusercontent.com/8436629/209313525-0edcf76a-14bd-43a9-9417-fbb0520be19a.png)

Unfortunately we don't have a linter for this, so we have to review formatting manually.

Reply:
@gavv When I add tab space golang linter (vscode) reverting back to single space. Also the above lines have single space too.

Reply:
Interesting. Well, I suppose there should be a way to configure vscode to disable this behavior. 

BTW this kind of formatting was not invented for me, this is how godoc wants to see code examples, so we have to use it.

Reply:
> Interesting. Well, I suppose there should be a way to configure vscode to disable this behavior.

Sure, yea I'll do that. 

> BTW this kind of formatting was not invented by me, this is how godoc wants to see code examples, so we have to use it.

Sure, ill make them tab space.